### ISSUE TITLE
button-name rule gives incorrect advise for button elements

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#1326]
Axe-core should not pass the following element:

```html
<button value="foo" tabindex="-1"></button>
```

This is caused by the `value` attribute. The button-name rule has the non-empty-value check for `input[type=button]` elements, and relies on `focusable-no-name` to still fail buttons if they have a value attribute but no accessible name. Firstly this leads to a confusing remediation message, where axe is saying to put a value attribute on buttons. Secondly, because `focusable-no-name` looks at focus, you can trip it up by putting `tabindex="-1"` on the button.

I think the way to fix this is to break the rule up. Have one rule for `button` elements, and a second for `input` buttons. This is similar to the problem we have with `svg` elements where we're recommending to put an `alt` attribute on them. See #1192

**Comment by straker** [comment]
As an alternate solution, could we make the `non-empty-value` check look at the `nodeName` of the element and make sure it's `input` to determine pass/fail?

**Comment by jeankaplansky** [comment]
https://dequeuniversity.com/rules/axe/3.3/button-name
https://dequeuniversity.com/rules/axe/3.3/input-button-name

**Comment by chandana7393** [comment]
Tested, working as expected.
![button_value_rule](https://user-images.githubusercontent.com/45285854/61214664-1f770380-a726-11e9-8c48-c0f0afd7311c.PNG)
**Tested Environment:**
Axe-coconut -3.8.1.19604v
Chrome - 75v
OS - Windows 10 64 bit.
